beacon-chain/rpc/service.go

```diff
@@ -31,13 +31,10 @@ func init() {
 }
 
 type chainService interface {
-	IncomingBlockFeed() *event.Feed
-	// These methods are not called on-demand by a validator
-	// but instead streamed to connected validators every
-	// time the canonical head changes in the chain service.
 	CanonicalBlockFeed() *event.Feed
-	CanonicalStateFeed() *event.Feed
 	StateInitializedFeed() *event.Feed
+	ReceiveBlock(block *pbp2p.BeaconBlock) (*pbp2p.BeaconState, error)
```

Comment:
Changed this interface from using event feeds to using direct calls to the receive block function